src/FileFormat/ExcelConverter.php

```diff
@@ -94,40 +72,47 @@ public function fromHtml($html, $selector = null)
 
-        return $this->getExcelService()->createPHPExcelObject($tmpFile['path']);
+        return IOFactory::load($tmpFile['path']);
         // tmpfile is deleted automatically
     }
 
     /**
      * Create response with excel download.
      *
-     * @param \PHPExcel $xlObj       Excel object to create the download from
-     * @param string    $filename    filename to give to the download
-     * @param string    $format      format of write (like Excel2007)
-     * @param string    $contentType mime content type
+     * @param PhpOffice\PhpSpreadsheet\Spreadsheet $spreadsheet Excel object to create the download from
+     * @param string                               $filename    filename to give to the download
+     * @param string                               $format      format of write (like Excel2007)
+     * @param string                               $contentType mime content type
      *
      * @return \Symfony\Component\HtmlFoundation\Response
      */
-    public function createResponse(\PHPExcel $xlObj, $filename, $format, $contentType)
+    public function createResponse(Spreadsheet $spreadsheet, $filename, $format, $contentType)
     {
-        return self::createExcelResponse($this->getExcelService(), $xlObj, $filename, $format, $contentType);
+        return self::createExcelResponse($spreadsheet, $filename, $format, $contentType);
     }
 
     /**
      * Create response with excel download.
      *
-     * @param Luiggio\ExcelBundle\Factory $excelSrvc
-     * @param \PHPExcel                   $xlObj       Excel object to create the download from
-     * @param string                      $filename    filename to give to the download
-     * @param string                      $format      format of write (like Excel2007)
-     * @param string                      $contentType mime content type
+     * @param PhpOffice\PhpSpreadsheet\Spreadsheet $spreadsheet Excel object to create the download from
+     * @param string                               $filename    filename to give to the download
+     * @param string                               $writerType  format of write (like Excel2007)
+     * @param string                               $contentType mime content type
      *
      * @return \Symfony\Component\HtmlFoundation\Response
      */
-    public static function createExcelResponse($excelSrvc, \PHPExcel $xlObj, $filename, $format, $contentType)
+    public static function createExcelResponse(Spreadsheet $spreadsheet, $filename, $writerType, $contentType)
     {
-        $xlWr = $excelSrvc->createWriter($xlObj, $format);
-        $response = $excelSrvc->createStreamedResponse($xlWr);
+        $writer = IOFactory::createWriter($spreadsheet, $writerType);
+
+        $response = new StreamedResponse(
+            function () use ($writer) {
+                $writer->save('php://output');
+            },
+            200,
+            []
```

Comment:
Ah, I see. The headers are set later. So it is fine. We could skip this arg entirely ([] is the default).